File: backend/wrapitup_planner/src/main/java/es/wrapitup/wrapitup_planner/dto/CalendarEventDTO.java
package es.wrapitup.wrapitup_planner.dto;

import es.wrapitup.wrapitup_planner.model.EventColor;
import java.time.LocalDateTime;
import lombok.Data;

@Data
public class CalendarEventDTO {
    private Long id;
    private Long userId;
    private String title;
    private String description;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private EventColor color;
    private String colorHex;
    private Boolean allDay;
    private LocalDateTime createdAt;
    private LocalDateTime lastModified;
    
    // Custom setter to maintain logic
    public void setColor(EventColor color) {
        this.color = color;
        if (color != null) {
            this.colorHex = color.getHexCode();
        }
    }
}

File: frontend/WrapItUp-Planner/src/app/dtos/calendar-event.dto.ts
import { EventColor } from './event-color.enum';

export interface CalendarEventDTO {
  id?: number;
  userId: number;
  title: string;
  description?: string;
  startDate: string;
  endDate: string;
  allDay: boolean;
  color: EventColor;
  colorHex?: string;
  createdAt?: string;
  updatedAt?: string;
}
